vms/platformvm/state/state.go

```diff
@@ -420,6 +429,18 @@ type txAndStatus struct {
 	status status.Status
 }
 
+type wrappedFxOwner struct {
+	owner fx.Owner
+	size  int
+}
+
+func wrappedFxOwnerSize(_ ids.ID, w *wrappedFxOwner) int {
+	if w == nil {
+		return ids.IDLen + constants.PointerOverhead
+	}
+	return ids.IDLen + w.size + constants.PointerOverhead
+}
```

Comment:
I understand this wrapper is just to be able to use fx.Owner in a sizedCache without recalculating the size every time.
The wrapper feels like a band-aid to me. I'd add a size method to fx.Owner interface to let ever owner be used in a sizedCache. Also fx.Owner implementation may be able to calculate statically their serialized size, without need to actually serialize data

Reply:
I didn't think it made sense to have a `Bytes()` like method for `fx.Owner` since the normal usage is passing it into another struct. This is a special case since we're explicitly tracking the owners separately so I'd like to keep it contained here

Reply:
I would prefer implementing this wrapper for now and if we find that we keep needing this in other places we can consider implementing it on the `fx` package. I like package being able to define their own abstractions and keeping the interface as minimal as possible